### ISSUE TITLE
Nice To Have :: Internal use of GuzzleHttp\RequestOptions

### ISSUE THREAD

**Issue by xbugster** [guzzle/guzzle#1302]
regards version: ~6.1.0
guzzlehttp/guzzle/src/Client.php lines 282-387 (at least) is using hard-coded names in my case:
form_params rather than GuzzleHttp\RequestOptions::FORM_PARAMS as documentation states.
same for other types like multipart, known headers and others.
**Important to state that it is nice to have the same compliance soft-wide.**

**Comment by mtdowling** [comment]
Sure. Care to send a pull request?

**Comment by xbugster** [comment]
Sure. in a day or two.

**Comment by xbugster** [comment]
Pull request issued.

**Comment by mtdowling** [comment]
Closing. Waiting on response to https://github.com/guzzle/guzzle/pull/1312
